### PR TITLE
Feat: list of pagination options based on the number of total elements

### PR DESCRIPTION
**Pull request by jasanhdz**
## Summary

Previously, the page component was putting a fixed number of options from which the list of elements was cut despite the fact that the total number of elements was less than that list, with this change what is done is that the list of options of cut are made from the number of totalRows so that it does not create more options than they need and in this way make the user see the maximum number of available elements and no more.

Screen Shot 2023-05-02 at 10 09 58 p m

## Task

- [LEN-649](https://dd360.atlassian.net/browse/LEN-649)

## Affected sections

```
src/components/Pagination/Pagination.test.tsx
src/components/Pagination/Pagination.tsx
src/stories/Pagination.stories.tsx
```

## How did you test this change?
Screen Shot 2023-05-02 at 10 07 26 p m

### CODE REVIEW

**Comment by tbor00** [comment]
Thanks for this PR 
I will review this 👀

File: src/components/Pagination/Pagination.tsx [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
+} from 'react'
 import { ChevronLeftIcon, ChevronRightIcon } from '@heroicons/react/outline'
 import { composeClasses } from 'lib/classes'
 import Text from '../Typography'
 
 export interface PaginationProps {
   totalPages: number
+  totalRows?: number
   currentPage: number
   sliceSize?:
```
Comment (AntonioFeregrino) [inline src/components/Pagination/Pagination.tsx:16]: Podríamos declarar un type, algo como `type Pages = | '5' | '10' | '15' | '20' | '25' | ...` 
 Para poder utilizarlo aquí 
 `sliceSize?: Pages`
 y aquí 
 `setSelectSliceSize(e.target.value as Pages)`

File: src/components/Pagination/Pagination.tsx [hunk 2]
```diff
@@ -75,24 +102,19 @@ const Pagination = ({
       <div className="flex items-center">
         {firstText && <Text size="base">{firstText}</Text>}
         <select
+          role="select-slice-size"
           name="custom-pagination"
           value={selectSliceSize}
           className="w-12 pl-2 mr-2 outline-none text-primary bg-transparent"
           onChange={(e) => handleChange(e)}
```
Comment (AntonioFeregrino) [inline src/components/Pagination/Pagination.tsx:109]:
```suggestion
          onChange={handleChange}
```

File: src/components/Pagination/Pagination.tsx [hunk 3]
```diff
@@ -33,6 +40,7 @@ const buttonStyle =
 
 const Pagination = ({
   totalPages,
```
Comment (AntonioFeregrino) [inline src/components/Pagination/Pagination.tsx:42]: El totalPages debería ser calculado por nosotros, no? Con el total de filas entre el tamaño seleccionado

Comment (jasanhdz) [reply src/components/Pagination/Pagination.tsx:42]: Si lo podriamos calcular, pero ya viene del back. incluso si cambia el slide, las props del request se actualizan y eso hace un refech y las props vuelven a cambiar con la información de la paginación actualizada.

**Review by lautarovilas — APPROVED** [review]

File: src/components/Pagination/Pagination.tsx [hunk 4]
```diff
@@ -75,24 +102,19 @@ const Pagination = ({
...
-          <option value="25">25</option>
-          <option value="30">30</option>
-          <option value="35">35</option>
-          <option value="40">40</option>
-          <option value="45">45</option>
-          <option value="50">50</option>
-          <option value="55">55</option>
-          <option value="60">60</option>
-          <option value="65">65</option>
+          {options.map((option) => {
```
Comment (tbor00) [inline src/components/Pagination/Pagination.tsx:111]: To avoid confusion

```suggestion
          {options.map((opt) => {
```

**Review by tbor00 — COMMENTED** [review]
I think the totalRows we can calculate with a simple math:

```ts
const totalRows = totalPages / sliceSize
```

And this now is dynamic and available for all use cases

**Pull request merged**